### ISSUE TITLE
Add to trained custom model without retraining whole set?

### ISSUE THREAD

**Issue by bertelschmitt** [AntonMu/TrainYourOwnYOLO#103]
Projects runs well and fast (on GPU). Thank you! How would I ADD images and tags to my trained custom model? I have thousands of images, and I would like to add a few hundred every so often without retraining the whole model (takes hours, even on GPU.)  Updates would contain new images with existing tags, and new images with new tags.

**Comment by AntonMu** [comment]
Hi @bertelschmitt - you are welcome. 

The code should work the very same way if you use your own pre-trained weights. So instead of loading the default YOLO weights you just load your already trained weights. The model is loaded [in line 160-162](https://github.com/AntonMu/TrainYourOwnYOLO/blob/master/2_Training/Train_YOLO.py#L160-L162  in the train script. You can also adjust the number of epochs as well as the number of layers to unfreeze. If you add new classes you may need to be a bit more careful to keep the numbering of the original classes unchanged.

**Comment by robisen1** [comment]
I have a dataset taken from a drone. The images are larger that  416 but still multiples of 32. How do I train on these new images and create new anchors?  Thank you.

**Comment by bertelschmitt** [comment]
Got it, thanks, I'll look into it

**Comment by shahzaibraza37** [comment]
@bertelschmitt Did you find any solution regarding this. 
I started the training on my previous model but When I do the detection, it does not have the old classes and trained data on the new generated weight file.

Has anyone has the solution regarding this issue?

**Comment by bertelschmitt** [comment]
@shahzaibraza37, I haven't looked into it yet. @AntonMu, could you please specify which file(s) to replace with which?  Thank you!

**Comment by robisen1** [comment]
> Projects runs well and fast (on GPU). Thank you! How would I ADD images and tags to my trained custom model? I have thousands of images, and I would like to add a few hundred every so often without retraining the whole model (takes hours, even on GPU.) Updates would contain new images with existing tags, and new images with new tags.

Transfer learning should work. Freeze your model, then reference your model in retraining with the new images to update it. You dont have to use your full training set. see https://towardsdatascience.com/a-comprehensive-hands-on-guide-to-transfer-learning-with-real-world-applications-in-deep-learning-212bf3b2f27a. That being said Alexy says you can also do fine-tuning instead of transfer see https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects: 

to speedup training (with decreasing detection accuracy) do Fine-Tuning instead of Transfer-Learning, set param stopbackward=1 here: https://github.com/AlexeyAB/darknet/blob/6d44529cf93211c319813c90e0c1adb34426abe5/cfg/yolov3.cfg#L548 then do this command: ./darknet partial cfg/yolov3.cfg yolov3.weights yolov3.conv.81 81 will be created file yolov3.conv.81, then train by using weights file yolov3.conv.81 instead of darknet53.conv.74

**Comment by shahzaibraza37** [comment]
@robisen1 But how can we do it for winodws. I did not find any command or way to do it with windows anaconda.

**Comment by MMaurits** [comment]
@shahzaibraza37 im also looking  for  an answer to the same question. Did you already found the answer?

**Comment by shahzaibraza37** [comment]
Hi @MMaurits,

I did find the answer in the research paper of  https://arxiv.org/pdf/1612.08242.pdf. As it is written 
> The high resolution classification
network of 448*448 gives us an increase of almost 4% mAP.
